tests/integration/kubernetes/confidential/k8s-confidential.bats

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+IMAGE_REPO="ghcr.io/confidential-containers/test-container"
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	( [ "${KATA_HYPERVISOR}" == "firecracker" ] || [ "${KATA_HYPERVISOR}" == "dragonball" ] || [ "${KATA_HYPERVISOR}" == "qemu-tdx" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ] ) && skip "Test not supported yet."
+
+	pod_name="confidential-unencrypted-test"
+	container_name="confidential-unencrypted-container"
+	get_pod_config_dir
+
+	k8s_generate_service_yaml "${TEST_DIR}/${pod_name}.yaml" "${IMAGE_REPO}:unencrypted"
+	k8s_yaml_set_annotation "${TEST_DIR}/${pod_name}.yaml" "io.katacontainers.config.pre_attestation.enabled" "false"
+}
+
+@test "Test unencrypted confidential container launch success." {
+  # Start the service/deployment/pod
+  esudo kubectl apply -f "${TEST_DIR}/sev-unencrypted.yaml"
+  
+  # Retrieve pod name, wait for it to come up, retrieve pod ip
+  pod_name=$(esudo kubectl get pod -o wide | grep ${pod_name} | awk '{print $1;}')
+
+  # Check pod creation
+  kubectl wait --for=condition=Ready --timeout=20 pod "$pod_name"
+
+  pod_ip=$(esudo kubectl get pod -o wide | grep ${pod_name} | awk '{print $6;}')
+
+  print_service_info
+
+  if [ "${KATA_HYPERVISOR}" == "qemu-snp" ]; then
+	# Look for SEV enabled in container dmesg output
+	sev_enabled=$(ssh -i ${SSH_KEY_FILE} \
+		-o "StrictHostKeyChecking no" \
+		-o "PasswordAuthentication=no" \
+		-t root@${pod_ip} \
```

Comment:
The SSH stuff is not platform specific. We can should be able to ssh into the image for any guest. It is only the command that is run via ssh that is platform specific. You might want to refactor this conditional to reflect that.

Reply:
I agree with Tobiun that the `ssh` part here is generic, and we can rely on an environment variable like `$searched_content`, where we can tie that `searched_content` to the machine where we're running the test on.

Reply:
Who's setting SSH_KEY_FILE?  And where?

Reply:
Made ssh function generic and added `DMESG_STRING` for content searching.

Reply:
Is every platform going to check dmesg for the status of CoCo? This might not be generic enough. Maybe we should make some kind of `execute_in_guest` function that takes any command and runs it via ssh. Then we could have a switch statement where each case calls that function with the appropriate command and checks the output. The `execute_in_guest` function would be useful in other tests as well. wdyt?